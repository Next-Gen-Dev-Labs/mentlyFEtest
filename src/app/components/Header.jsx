import React from 'react';
import { Bell, MessageSquare, Menu, X } from 'lucide-react';
import Image from 'next/image';
import user_header_logo from '../../../public/user-header.png'; 
import techrity_logo from '../../../public/techrity-logo.png'; // Techrity Logo Image

export default function Header({ isSidebarOpen, onToggleSidebar }) {
  return (
    <header className="bg-white shadow-sm z-10 relative">
      <div className="flex items-center justify-between px-4 py-4 md:py-2 md:px-6 ">
        {/* Sidebar Toggle - Mobile Only */}
        <div className="md:hidden">
          <button 
            onClick={onToggleSidebar} 
            className="text-purple-800 bg-purple-100 p-2 rounded-md shadow-sm"
            aria-label={isSidebarOpen ? "Close sidebar" : "Open sidebar"}
          >
            {isSidebarOpen ? <X size={18} /> : <Menu size={18} />}
          </button>
        </div>

        {/* Logo and Title for larger screens */}
        <div className="flex items-center space-x-2 md:space-x-4">
          {/* Techrity Logo (Always visible) */}
          <div className="h-8 w-24 relative md:flex items-center">
            <Image 
              src={techrity_logo} 
              alt="Techrity Logo" 
              width={100} 
              height={32} 
              className="object-contain"
              priority
            />
          </div>
          























































































          
        </div>

        {/* Right Side Icons/User Info */}
        <div className="flex items-center space-x-3 md:space-x-4 ml-auto">
          {/* Notification Bell */}
          <button className="p-2 rounded-full hover:bg-gray-100 relative">
            <Bell size={20} className="text-gray-600" />
            <div className="absolute top-1 right-1 w-1.5 h-1.5 bg-red-500 rounded-full"></div>
          </button>

          {/* Chat Icon - Hide on Mobile */}
          <button className="p-1 rounded-full bg-purple-100 text-purple-700 hidden sm:inline-flex">
            <MessageSquare size={16} />
          </button>

          {/* User Info */}
          <div className="flex items-center space-x-2">
            <div className="w-8 h-8 rounded-full bg-[#8B72FC] flex items-center justify-center text-white">
              <Image 
                src={user_header_logo}
                alt="User Avatar"
                width={40}
                height={40}
                className="rounded-full object-cover"
              />
            </div>
            {/* Hide User Text on Mobile */}
            <div className="text-sm hidden sm:block">
              <div className="text-black font-medium leading-tight">Techrity Foundation</div>
              <div className="text-xs text-gray-500">Member</div>
            </div>
          </div>
        </div>
      </div>
    </header>
  );
}
