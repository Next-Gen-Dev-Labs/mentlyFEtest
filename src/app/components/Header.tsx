"use client"
import { Icon } from "@iconify/react";
import Image from "next/image";
import Link from "next/link";
import { useAppContext } from "../context/AppContext";

const Header = () => {
    const { isSidebarOpen, setIsSidebarOpen, darktheme} = useAppContext()
    return (
        <div className={`w-full h-full flex gap-2 justify-between items-center p-4  ${darktheme ? "bg-appDeepTextBlue" : "bg-[#FDFDFD]" }`}>
            <button
                aria-expanded={isSidebarOpen}
                aria-controls="mobile-menu"
                onClick={() => setIsSidebarOpen(true)}
                
                className={`lg:invisible ${isSidebarOpen ? "invisible" : "visible"} lg:pointer-events-none cursor-pointer relative left-0`}
                >
                <Icon icon="solar:hamburger-menu-broken" className="lg:invisible lg:pointer-events-none  cursor-pointer " width="30" height="30" aria-label="Open menu" />
            </button>
            <span className="flex gap-2">
                <Image src="notifications.svg" alt="notification icon" width={20} height={20} />
                <div className="flex gap-2 items-center h-full lg:mr-10">
                    <Image src="/profile.png" alt="profile icon" width={50} height={50} className="rounded-full" />
                    <div>
                        <p className={`${darktheme ? "text-white" : "text-[#404040] text-sm"}`}>Godwin Jimmy</p>
                        <div className="flex gap-2">
                            <p className={`${darktheme ? "text-white" : "text-[#777795]"} text-xs`}>Free Plan</p>
                            <Link className="text-[#2B85FE] font-bold text-sm" href="/">Upgrade</Link>
                        </div>
                    </div>
                    <Icon icon="mingcute:down-line" className="text-[#C2C2C2] cursor-pointer" width="24" height="24" />
                </div>
            </span>
        </div>
    );
}

export default Header;