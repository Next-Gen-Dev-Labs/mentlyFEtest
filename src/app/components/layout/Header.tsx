"use client";

import { useState, useEffect, useRef } from "react";
import { AnimatePresence, motion } from "framer-motion";
import Image from "next/image";
import Logo from "../../../../public/images/icons/HeaderLogo.svg";
import MenuIcon from "../../../../public/images/icons/menuIcon.svg";
import BellIcon from "../../../../public/images/icons/bellIcon.svg";
import { BsChevronDown } from "react-icons/bs";

interface HeaderProps {
  toggleSidebar: () => void;
  isSidebarExpanded: boolean;
}

export default function Header({
  toggleSidebar,
  isSidebarExpanded,
}: HeaderProps) {
  const [isNotificationOpen, setIsNotificationOpen] = useState(false);
  const [isAtTop, setIsAtTop] = useState(true);
  const mainContentRef = useRef<HTMLElement | null>(null);

  // Close notifications when clicking outside
  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      const target = event.target as HTMLElement;
      if (
        isNotificationOpen &&
        !target.closest("[data-notification-container]")
      ) {
        setIsNotificationOpen(false);
      }
    };

    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, [isNotificationOpen]);

  // Find main content element and track scroll position
  useEffect(() => {
    mainContentRef.current = document.querySelector("main");

    const handleScroll = () => {
      if (mainContentRef.current) {
        // For the main element's scroll position
        setIsAtTop(mainContentRef.current.scrollTop <= 20);
      } else {
        // Fallback to window scroll
        setIsAtTop(window.scrollY <= 20);
      }
    };

    // Initial check
    handleScroll();

    // Add event listener to the right element
    const scrollElement = mainContentRef.current || window;
    scrollElement.addEventListener("scroll", handleScroll, { passive: true });

    return () => {
      if (mainContentRef.current) {
        mainContentRef.current.removeEventListener("scroll", handleScroll);
      } else {
        window.removeEventListener("scroll", handleScroll);
      }
    };
  }, []);

  const handleScrollToggle = () => {
    const mainContent =
      mainContentRef.current || document.querySelector("main");

    if (mainContent) {
      if (isAtTop) {
        mainContent.scrollTo({
          top: mainContent.scrollHeight,
          behavior: "smooth",
        });
      } else {
        mainContent.scrollTo({
          top: 0,
          behavior: "smooth",
        });
      }
    } else {
      // Fallback to window scroll
      if (isAtTop) {
        window.scrollTo({
          top: document.body.scrollHeight,
          behavior: "smooth",
        });
      } else {
        window.scrollTo({
          top: 0,
          behavior: "smooth",
        });
      }
    }
  };
  return (
    <header className="bg-white border-b border-gray-200 py-4 px-4 md:px-6 sticky top-0 z-20 shadow-sm">
      <div className="flex justify-between items-center max-w-[1920px] mx-auto">
        {/* Left side - Menu toggle */}
        <div className="flex items-center md:hidden">
          <motion.button
            whileHover={{ scale: 1.05 }}
            whileTap={{ scale: 0.95 }}
            onClick={toggleSidebar}
            className="p-2 mr-2 text-gray-600 hover:bg-gray-100 rounded-full transition-all duration-200"
            aria-label={
              isSidebarExpanded ? "Collapse sidebar" : "Expand sidebar"
            }
          >
            <motion.div
              animate={{ rotate: isSidebarExpanded ? 180 : 0 }}
              transition={{ duration: 0.8 }}
            >
              <Image
                src={MenuIcon}
                alt="Toggle menu"
                width={24}
                height={24}
                className="text-gray-600"
              />
            </motion.div>
          </motion.button>
        </div>

        {/* Right side - Notifications and profile */}
        <div className="flex items-center space-x-4 ml-auto">
          <motion.div className="relative" data-notification-container>
            <motion.button
              whileHover={{ scale: 1.05 }}
              whileTap={{ scale: 0.95 }}
              className="relative p-2 rounded-full text-gray-700 hover:bg-gray-100 transition-all duration-200"
              onClick={() => setIsNotificationOpen(!isNotificationOpen)}
              aria-label="Notifications"
              aria-expanded={isNotificationOpen}
              aria-controls="notification-dropdown"
            >
              <Image
                src={BellIcon}
                alt="Notifications"
                width={24}
                height={24}
                className="h-6 w-6"
              />
              <span className="absolute top-1 right-1 h-3 w-3 bg-red-500 rounded-full border-2 border-white"></span>
            </motion.button>

            {/* Notification dropdown */}
            <AnimatePresence>
              {isNotificationOpen && (
                <motion.div
                  id="notification-dropdown"
                  initial={{ opacity: 0, y: -10, scale: 0.95 }}
                  animate={{ opacity: 1, y: 0, scale: 1 }}
                  exit={{ opacity: 0, y: -10, scale: 0.95 }}
                  transition={{ type: "spring", stiffness: 400, damping: 30 }}
                  className="absolute right-0 mt-2 w-80 bg-white shadow-lg rounded-lg overflow-hidden z-50 border border-gray-200"
                  style={{ transformOrigin: "top right" }}
                >
                  <div className="p-4 border-b border-gray-100 flex justify-between items-center">
                    <h3 className="font-semibold text-gray-800">
                      Notifications
                    </h3>
                    <span className="text-xs bg-purple-100 text-purple-800 px-2 py-1 rounded-full">
                      3 New
                    </span>
                  </div>
                  <div className="max-h-96 overflow-y-auto">
                    <motion.div
                      className="p-4 border-b border-gray-100 hover:bg-gray-50 cursor-pointer"
                      whileHover={{ backgroundColor: "rgba(243, 244, 246, 1)" }}
                    >
                      <div className="flex items-start">
                        <div className="h-2 w-2 mt-1.5 bg-red-500 rounded-full mr-2 flex-shrink-0"></div>
                        <div>
                          <p className="text-sm font-medium text-gray-800">
                            New program published
                          </p>
                          <p className="text-xs text-gray-500 mt-1">
                            2 minutes ago
                          </p>
                        </div>
                      </div>
                    </motion.div>
                    <motion.div
                      className="p-4 border-b border-gray-100 hover:bg-gray-50 cursor-pointer"
                      whileHover={{ backgroundColor: "rgba(243, 244, 246, 1)" }}
                    >
                      <div className="flex items-start">
                        <div className="h-2 w-2 mt-1.5 bg-red-500 rounded-full mr-2 flex-shrink-0"></div>
                        <div>
                          <p className="text-sm font-medium text-gray-800">
                            New application received
                          </p>
                          <p className="text-xs text-gray-500 mt-1">
                            1 hour ago
                          </p>
                        </div>
                      </div>
                    </motion.div>
                    <motion.div
                      className="p-4 hover:bg-gray-50 cursor-pointer"
                      whileHover={{ backgroundColor: "rgba(243, 244, 246, 1)" }}
                    >
                      <p className="text-sm font-medium text-gray-800">
                        System maintenance scheduled
                      </p>
                      <p className="text-xs text-gray-500 mt-1">
                        Tomorrow at 2:00 AM
                      </p>
                    </motion.div>
                  </div>
                  <div className="p-3 border-t border-gray-100 text-center bg-gray-50">
                    <button className="text-sm text-purple-600 hover:text-purple-800 font-medium transition-colors duration-200">
                      View all notifications
                    </button>
                  </div>
                </motion.div>
              )}
            </AnimatePresence>
          </motion.div>

          {/* Profile */}
          <div className="flex items-center space-x-3">
            <div className="h-10 w-10 rounded-full bg-[#8b72fc] flex items-center justify-center overflow-hidden">
              <Image
                src={Logo}
                alt="Techrity logo"
                className="h-auto w-auto p-1"
                width={32}
                height={32}
              />
            </div>
            <motion.div
              className="hidden md:block"
              initial={{ opacity: 0, x: -5 }}
              animate={{ opacity: 1, x: 0 }}
              transition={{ delay: 0.1 }}
            >
              <p className="text-sm font-medium text-gray-800">
                Techrity Foundation
              </p>
              <p className="text-xs text-gray-500">Member</p>
            </motion.div>

            {/* Scroll button with animation */}
            <motion.button
              className="p-2 bg-[#6F01D0] text-white rounded-md hover:bg-purple-700"
              onClick={handleScrollToggle}
              whileHover={{ scale: 1.05 }}
              whileTap={{ scale: 0.95 }}
              aria-label={isAtTop ? "Scroll to bottom" : "Scroll to top"}
            >
              <motion.div
                animate={{ rotate: isAtTop ? 0 : 180 }}
                transition={{ duration: 0.4 }}
              >
                <BsChevronDown className="h-5 w-5" />
              </motion.div>
            </motion.button>
          </div>
        </div>
      </div>
    </header>
  );
}

// <div className="flex items-center space-x-3">
//   <button
//     className="hidden lg:block p-2 text-gray-400 hover:text-gray-600"
//     aria-label="List view"
//   >
//     <svg
//       className="h-5 w-5"
//       fill="none"
//       stroke="currentColor"
//       viewBox="0 0 24 24"
//     >
//       <path
//         strokeLinecap="round"
//         strokeLinejoin="round"
//         strokeWidth={2}
//         d="M4 6h16M4 12h16M4 18h16"
//       />
//     </svg>
//   </button>

//   <button
//     className="hidden lg:block p-2 text-gray-700"
//     aria-label="Grid view"
//   >
//     <svg
//       className="h-5 w-5"
//       fill="none"
//       stroke="currentColor"
//       viewBox="0 0 24 24"
//     >
//       <path
//         strokeLinecap="round"
//         strokeLinejoin="round"
//         strokeWidth={2}
//         d="M4 6h16M4 10h16M4 14h16M4 18h16"
//       />
//     </svg>
//   </button>

//   <motion.button
//     whileHover={{ scale: 1.05 }}
//     whileTap={{ scale: 0.95 }}
//     className="hidden md:flex bg-purple-50 text-purple-700 py-2 px-4 rounded-md items-center"
//     onClick={onManageWidgets}
//     aria-label="Manage widgets"
//   >
//     <svg
//       className="h-5 w-5 mr-2"
//       fill="none"
//       stroke="currentColor"
//       viewBox="0 0 24 24"
//     >
//       <path
//         strokeLinecap="round"
//         strokeLinejoin="round"
//         strokeWidth={2}
//         d="M12 4v16m8-8H4"
//       />
//     </svg>
//     Manage Widgets
//   </motion.button>

//   <motion.button
//     whileHover={{ scale: 1.05 }}
//     whileTap={{ scale: 0.95 }}
//     className="md:hidden p-2 rounded-full bg-purple-100 text-purple-700"
//     onClick={onManageWidgets}
//     aria-label="Manage widgets"
//   >
//     <svg
//       className="h-5 w-5"
//       fill="none"
//       stroke="currentColor"
//       viewBox="0 0 24 24"
//     >
//       <path
//         strokeLinecap="round"
//         strokeLinejoin="round"
//         strokeWidth={2}
//         d="M12 4v16m8-8H4"
//       />
//     </svg>
//   </motion.button>
// </div>;
