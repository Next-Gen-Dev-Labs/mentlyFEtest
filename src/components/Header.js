import { useState, useRef, useEffect } from "react";
import Image from "next/image";
import { ArrowDown2, Notification } from "iconsax-react";

const Header = ({ theme }) => {
  const [dropdownOpen, setDropdownOpen] = useState(false);
  const dropdownRef = useRef(null);


  useEffect(() => {
    function handleClickOutside(event) {
      if (dropdownRef.current && !dropdownRef.current.contains(event.target)) {
        setDropdownOpen(false);
      }
    }
    document.addEventListener("mousedown", handleClickOutside);
    return () => document.removeEventListener("mousedown", handleClickOutside);
  }, []);

  return (
    <header className="flex justify-end bg-white dark:bg-gray-700 p-4 shadow-[0px_2px_4px_0px_#00000010] fixed top-0 left-0 right-0 z-30 h-[70px] md:h-[82px]">
     
     <div className="flex items-center gap-6 md:gap-10 mr-10px] md:mr-[100px]">

        
        <div className="relative cursor-pointer">
          <Notification size="24" color={theme === "dark" ? "#fff" : "#4B5563"} />
          <span className="absolute top-0 right-0 bg-red-500 w-2.5 h-2.5 rounded-full"></span>
        </div>

        {/* User Profile Section */}
        <div className="relative">
          <div
            className="flex items-center gap-4"
          
          >
            <div className="flex gap-2"><Image
              src="/images/user-image.svg"
              alt="User Avatar"
              width={44}
              height={44}
              className="rounded-full"
            />
            <div>
              <span className="text-gray-700 dark:text-white font-semibold block">
                Godwin Jimmy
              </span>
              <span className="text-sm text-gray-500 dark:text-gray-400">
                Free Plan <a href="#" className="text-blue-500 font-semibold">Upgrade</a>
              </span>
            </div></div>
            
            {/* Dropdown Icon (Rotates when Open) */}
            <ArrowDown2 size="18" color="#C2C2C2"
              className={`transition-transform ${dropdownOpen ? "rotate-180" : "rotate-0"} cursor-pointer`}  ref={dropdownRef}  onClick={() => setDropdownOpen(!dropdownOpen)}
            />
          </div>

          {/* Dropdown Menu */}
          {dropdownOpen && (
            <div
              className="absolute right-0 mt-2 w-48 bg-white dark:bg-gray-800 shadow-lg rounded-lg py-2 transform transition-all duration-200 ease-out scale-95 opacity-0 animate-fadeInSlide"
            >
              <a href="/profile" className="block px-4 py-2 text-gray-700 dark:text-white hover:bg-gray-100 dark:hover:bg-gray-600">My Profile</a>
              <a href="/awards" className="block px-4 py-2 text-gray-700 dark:text-white hover:bg-gray-100 dark:hover:bg-gray-600">My Awards</a>
              <a href="/settings" className="block px-4 py-2 text-gray-700 dark:text-white hover:bg-gray-100 dark:hover:bg-gray-600">Settings</a>
              <a href="/logout" className="block px-4 py-2 text-red-600 hover:bg-gray-100 dark:hover:bg-gray-600">Logout</a>
            </div>
          )}
        </div>
      </div>
    </header>
  );
};

export default Header;
