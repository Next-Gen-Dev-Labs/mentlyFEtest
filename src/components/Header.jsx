'use client';

import { useSidebar } from '@/hooks/useSidebar'; // Import the custom hook

const Header = () => {
  const { toggleSidebar } = useSidebar(); // Use the hook to get the toggle function

  const handleClick = () => {
    console.log('Sidebar toggle clicked'); // for debugging
    toggleSidebar(); // Call the toggle function from the hook
  };

  return (
    <header className="bg-white shadow px-4 py-4 flex items-center justify-between md:justify-end">
      {/* Mobile hamburger */}
      <button
        onClick={handleClick}
        className="md:hidden text-[#1F0544] text-2xl"
      >
        <i className="bi bi-list"></i>
      </button>

      <div className="flex-1 flex justify-end items-center space-x-4">
        <i className="fas fa-bell text-gray-500 text-lg"></i>
        <div className="flex items-center space-x-2">
          <div className="w-8 h-8 rounded-full bg-purple-500 flex items-center justify-center text-white">
            <i className="fas fa-user"></i>
          </div>
          <div className="hidden sm:block">
            <div className="text-sm font-medium">Techrity Foundation</div>
            <div className="text-xs text-gray-500">Member</div>
          </div>
        </div>
      </div>
    </header>
  );
};

export default Header;
