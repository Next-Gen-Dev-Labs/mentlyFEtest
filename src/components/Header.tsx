"use client";
import Image from "next/image";
import { BellIcon } from "@heroicons/react/24/outline";
import { ArrowDown2 } from "iconsax-react";

const Header = () => {
  return (
    <header className="top-0 left-0 w-full bg-white shadow-md flex justify-between items-center px-6 md:px-10 h-[82px]">
      {/* Left Section: Logo or Placeholder */}
      <div className="flex items-center">
        <Image src="/man.png" alt="Logo" width={40} height={40} className="hidden md:block" />
      </div>

      {/* Right Section: Notification & Profile */}
      <div className="flex items-center space-x-4">
        {/* Notification Icon */}
        <div className="relative ">
          <BellIcon className="h-6 w-6 text-gray-500 cursor-pointer" />
          <span className="absolute top-0 right-0 h-2 w-2 bg-red-500 rounded-full"></span>
        </div>

        {/* Profile */}
        <div className=" flex items-center space-x-3 cursor-pointer">
          <Image
            src="/man.png"
            alt="Profile"
            width={40}
            height={40}
            className="rounded-full"
          />
          <div>
            <p className="text-sm font-medium text-gray-900">Godwin Jimmy</p>
            <p className="text-xs text-gray-500">
              Free Plan <span className="text-blue-600 cursor-pointer">Upgrade</span>
            </p>
          </div>
        </div>

        <div>
          <ArrowDown2 color="grey" size="15" />
        </div>
      </div>
    </header>
  );
};

export default Header;
