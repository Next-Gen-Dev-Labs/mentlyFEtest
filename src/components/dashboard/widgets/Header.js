'use client';
import "../lib/fontawesome";

const Header = ({ onToggleSidebar }) => {
  return (
    <header className="bg-white shadow px-4 py-4 flex items-center justify-between md:justify-end">
      <button
        onClick={onToggleSidebar}
        className="md:hidden text-[#1F0544] text-2xl"
      >
        <i className="bi bi-list"></i>
      </button>

      <div className="flex-1 flex justify-end items-center space-x-4">
        <i className="fas fa-bell text-gray-500 text-lg"></i>
        <div className="flex items-center space-x-2">
          <div className="w-8 h-8 rounded-full bg-purple-500 flex items-center justify-center text-white">
            <i className="fas fa-user"></i>
          </div>
          <div className="hidden sm:block">
            <div className="text-sm font-medium">Techrity Foundation</div>
            <div className="text-xs text-gray-500">Member</div>
          </div>
        </div>
      </div>
    </header>
  );
};

export default Header;
