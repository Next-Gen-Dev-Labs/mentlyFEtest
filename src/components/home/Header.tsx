"use client";
import React, { useState } from "react";
import SlideNav from "./SlideNav";

const Header = () => {
  const [open, setOpen] = useState(false);

  const toggle = () => {
    setOpen(!open);
  };

  const close = () => {
    setOpen(false);
  };
  return (
    <div className="w-full">
      {/* manage widget */}
      <div className="w-full grid place-content-end">
        <div className="flex items-center gap-7">
          {/* image 1 */}
          <div className="">
            <button className="cursor-pointer">
              <img src="/widget.svg" alt="" />
            </button>
          </div>

          {/* image 2 */}
          <div className="">
            <button className="cursor-pointer">
              <img src="/widget.png" alt="" />
            </button>
          </div>

          {/* text */}
          <button
            className="text-[#1F0954] font-bold cursor-pointer"
            onClick={toggle}
          >
            Manage Widgets
          </button>
        </div>
      </div>
      {/* onboard message */}
      <div className="w-full">
        <div className="bg-blue min-h-[61px] px-7 rounded-[4px] text-white flex items-center justify-between gap-4 flex-wrap py-2">
          <div className="flex lg:items-center flex-col lg:flex-row xl:gap-20 flex-wrap">
            <p className="text-sm lg:text-base xl:text-[26.21px] font-semibold">
              Welcome Aboard, Blessing ðŸ‘‹
            </p>
            <p className="text-sm lg:text-base xl:text-[20px] font-semibold text-[#BDBDBD]">
              Weâ€™re thrilled to have you join Techrity Team!
            </p>
          </div>
          {/* update button */}
          <div className="">
            <button className="btn-up text-[#1F0954] font-bold">
              Update Profile
            </button>
          </div>
        </div>
      </div>

      {/* side bar */}
      <div>
        <SlideNav open={open} close={close} />
      </div>
    </div>
  );
};

export default Header;
