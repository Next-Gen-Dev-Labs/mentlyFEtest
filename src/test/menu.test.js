import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '@/components/reusables/menu';

// Mock next/image
jest.mock('next/image', () => ({
  __esModule: true,
  default: ({ src, alt, className, 'data-testid': testId }) => (
    <img 
      src={src} 
      alt={alt} 
      className={className}
      data-testid={testId || `image-${alt.toLowerCase().replace(/\s+/g, '-')}`}
    />
  ),
}));

// Mock SVG imports
jest.mock('@/assets/icons/Vector (3).svg', () => 'vector3-icon');
jest.mock('@/assets/icons/Vector (1).svg', () => 'vector1-icon');

describe('Header Component', () => {
  const testProps = {
    head: 'Test Header',
  };

  it('renders correctly with given props', () => {
    render(<Header {...testProps} />);
    
    // Check header text
    expect(screen.getByText(testProps.head)).toBeInTheDocument();
    expect(screen.getByText(testProps.head)).toHaveClass('text-[#B0B0B0]');
    expect(screen.getByText(testProps.head)).toHaveClass('font-bold');
    
    // Check "See all" button
    const seeAllButton = screen.getByText('See all');
    expect(seeAllButton).toBeInTheDocument();
    expect(seeAllButton).toHaveClass('text-[#6F01D0]');
    expect(seeAllButton).toHaveClass('hover:text-[#6f01d0d1]');
    expect(seeAllButton).toHaveClass('font-semibold');
  });

  it('renders all icons correctly', () => {
    render(<Header {...testProps} />);
    
    // Menu icon (using explicit test ID)
    const menuIcon = screen.getByTestId('menu-icon');
    expect(menuIcon).toBeInTheDocument();
    expect(menuIcon).toHaveAttribute('src', 'vector1-icon');
    expect(menuIcon).toHaveClass('h-4');
    expect(menuIcon).toHaveClass('w-4');
    
    // Action icon (using explicit test ID)
    const actionIcon = screen.getByTestId('action-icon');
    expect(actionIcon).toBeInTheDocument();
    expect(actionIcon).toHaveAttribute('src', 'vector3-icon'); // Fixed expectation
    expect(actionIcon).toHaveClass('cursor-pointer');
  });

  it('has correct layout structure', () => {
    const { container } = render(<Header {...testProps} />);
    
    // Check main container
    const headerDiv = container.firstChild;
    expect(headerDiv).toHaveClass('flex');
    expect(headerDiv).toHaveClass('justify-between');
    expect(headerDiv).toHaveClass('items-center');
    
    // Check left section
    const leftSection = screen.getByText(testProps.head).parentElement;
    expect(leftSection).toHaveClass('flex');
    expect(leftSection).toHaveClass('gap-2.5');
    expect(leftSection).toHaveClass('items-center');
    
    // Check right section
    const rightSection = screen.getByText('See all').parentElement;
    expect(rightSection).toHaveClass('flex');
    expect(rightSection).toHaveClass('gap-5');
    expect(rightSection).toHaveClass('items-center');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Header {...testProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('applies hover styles correctly', () => {
    render(<Header {...testProps} />);
    const seeAllButton = screen.getByText('See all');
    expect(seeAllButton).toHaveClass('hover:text-[#6f01d0d1]');
  });

  it('handles icon clicks when provided', () => {
    // Currently no onClick handlers, but this is how you'd test them:
    // const mockClick = jest.fn();
    // render(<Header {...testProps} onMenuClick={mockClick} />);
    // fireEvent.click(screen.getByTestId('image-menu-icon'));
    // expect(mockClick).toHaveBeenCalled();
  });
});