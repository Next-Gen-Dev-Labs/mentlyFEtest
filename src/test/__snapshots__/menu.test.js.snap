// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex justify-between items-center"
  >
    <div
      class="flex gap-2.5 items-center"
    >
      <img
        alt="menu icon"
        class="h-4 w-4"
        data-testid="menu-icon"
        src="vector1-icon"
      />
      <h3
        class="font-bold text-[#B0B0B0]"
      >
        Test Header
      </h3>
    </div>
    <div
      class="flex gap-5 items-center"
    >
      <button
        class="text-sm text-[#6F01D0] hover:text-[#6f01d0d1] font-semibold cursor-pointer"
      >
        See all
      </button>
      <img
        alt="icon"
        class="cursor-pointer"
        data-testid="action-icon"
        src="vector1-icon"
      />
    </div>
  </div>
</DocumentFragment>
`;
